docs/examples/browser-devenv-v3/files/lib.js

```diff
@@ -0,0 +1,130 @@
+async function updateSite(filename, content){
+    const resp = await fetch(`${window.origin}/${filename}`, {method: 'put', body: content})
+    const newLocation = resp.headers.get('location')
+    window.location = new URL(newLocation).origin
+}
+
+async function publishSite(){
+    let resp = await fetch('ipns://localhost/?key=mysite', {method: 'POST'})
+    const key = resp.headers.get('location')
+    resp = await fetch(key, {method: 'POST', body: window.origin})
+    window.location = new URL(resp.headers.get('location')).origin
+}
+
+async function loadFile(filename){
+    const resp = await fetch(filename)
+    const content = await resp.text()
+    document.getElementById('idFilenameInput').value = filename
+//- document.getElementById('idContentInput').value = content
+    window.editor.setValue(content)                                          //+
+    if (filename.match(/\.js/)){                                             //+
+        editor.session.setMode("ace/mode/javascript")                        //+
+    } else if (filename.match(/\.html/)){                                    //+
+        editor.session.setMode("ace/mode/html")                              //+
+    }                                                                        //+
```

Comment:
Should this be uncommented and should there be an else to set the mode the plain text?

Reply:
Not sure I understand? The comments at the right is to indicate the lines that's been added.

Wrt to the else: I'll add that. Currently selecting a js/html file and then something else will cause the editor to still be in js/html mode

Reply:
My bad, misread the diff format in the diff view